src/backend/core/api/permissions.py

```diff
@@ -50,5 +53,21 @@ class AccessPermission(IsAuthenticated):
 
     def has_object_permission(self, request, view, obj):
         """Check permission for a given object."""
+        if isinstance(obj, models.Invitation):
+            team = get_object_or_404(models.Team, id=view.kwargs["team_id"])
+            return team.get_abilities(request.user).get("manage_accesses")
+
         abilities = obj.get_abilities(request.user)
         return abilities.get(request.method.lower(), False)
+
+
+class RelatedTeamAccessPermission(IsAuthenticated):
```

Comment:
Naming could be improved, to `RelatedTeamPermission` or something else. Reading it first, it feels the class has the responsibility to handle permissions for `TeamAccesses` only, when the doc states it's for team accesses and invitations.